sn_interface/src/network_knowledge/sections_dag.rs

```diff
@@ -0,0 +1,888 @@
+// Copyright 2022 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+#![allow(dead_code)]
+
+use super::errors::{Error, Result};
+use crdts::{
+    merkle_reg::{Hash, MerkleReg, Node, Sha3Hash},
+    CmRDT,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::{self, Debug, Formatter};
+use tiny_keccak::{Hasher, Sha3};
+
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+pub struct SectionInfo {
+    key: bls::PublicKey,
+    sig: bls::Signature,
+}
+
+impl Sha3Hash for SectionInfo {
+    fn hash(&self, hasher: &mut Sha3) {
+        let bytes: Vec<u8> = self
+            .key
+            .to_bytes()
+            .into_iter()
+            .chain(self.sig.to_bytes().into_iter())
+            .collect();
+        hasher.update(bytes.as_slice());
```

Comment:
You don't want to build up a bytes buffer and pass it all at once to hasher, instead call hasher update with each chunk of data you want to include in the hash.

```suggestion
        hasher.update(&self.key.to_bytes());
        hasher.update(&self.sig.to_bytes());
```

Reply:
Got it! Making the change!